Sources/SwiftParser/Types.swift

```diff
@@ -255,45 +263,22 @@ extension Parser {
     )
   }
 
-  @_spi(RawSyntax)
-  public mutating func parseTypeIdentifier(
-    stopAtFirstPeriod: Bool = false
-  ) -> RawTypeSyntax {
-    if self.at(.anyKeyword) && !stopAtFirstPeriod {
-      return RawTypeSyntax(self.parseAnyType())
+  mutating func parseTypeNameWithGenerics() -> (RawTokenSyntax, RawGenericArgumentClauseSyntax?) {
+    let (name, _) = self.parseDeclNameRef()
+    if self.atContextualPunctuator("<") {
+      return (name, self.parseGenericArguments())
     }
+    return (name, nil)
+  }
 
-    var result: RawTypeSyntax?
-    var keepGoing: RawTokenSyntax? = nil
-    var loopProgress = LoopProgressCondition()
-    repeat {
-      let (name, _) = self.parseDeclNameRef()
-      let generics: RawGenericArgumentClauseSyntax?
-      if self.atContextualPunctuator("<") {
-        generics = self.parseGenericArguments()
-      } else {
-        generics = nil
-      }
-      if let keepGoing = keepGoing {
-        result = RawTypeSyntax(RawMemberTypeIdentifierSyntax(
-          baseType: result!,
-          period: keepGoing,
-          name: name,
-          genericArgumentClause: generics,
-          arena: self.arena))
-      } else {
-        result = RawTypeSyntax(RawSimpleTypeIdentifierSyntax(
-          name: name, genericArgumentClause: generics, arena: self.arena))
-      }
-
-      if stopAtFirstPeriod {
-        keepGoing = nil
-      } else {
-        keepGoing = self.consume(if: .period) ?? self.consume(if: .prefixPeriod)
-      }
-    } while keepGoing != nil && loopProgress.evaluate(currentToken)
+  mutating func parseTypeIdentifier() -> RawTypeSyntax {
+    if self.at(.anyKeyword) {
```

Comment:
Do we also need to handle `some` here?

Reply:
I don't think so, but this seems a little odd to me actually - this `Any` not `any` in which case is `Any` really a keyword in this context (ends up being put into a simple type identifier as `anyKeyword`)? Also, it seems like it would be better to still parse generics in this context for recovery purposes - thoughts?

Reply:
Oh, maybe this should be called `captialAnyKeyword`. But nothing for this PR.

Makes sense to me. I won’t stop you from implementing better recovery.